src/org/rascalmpl/library/util/PathConfig.java

```diff
@@ -31,64 +32,49 @@
 import io.usethesource.vallang.ISourceLocation;
 import io.usethesource.vallang.IValue;
 import io.usethesource.vallang.IValueFactory;
-import io.usethesource.vallang.IWithKeywordParameters;
-import io.usethesource.vallang.exceptions.FactTypeUseException;
-import io.usethesource.vallang.io.StandardTextReader;
 import io.usethesource.vallang.type.Type;
 import io.usethesource.vallang.type.TypeFactory;
 import io.usethesource.vallang.type.TypeStore;
 
 public class PathConfig {
-	
 	private static final IValueFactory vf = ValueFactoryFactory.getValueFactory();
 	private final TypeFactory tf = TypeFactory.getInstance();
 	private final TypeStore store = new TypeStore();
 	
-	// WARNING: these definitions must reflect the definitions in util::Reflective.rsc
+	// WARNING: these definitions must reflect the definitions in `util::Reflective`
 	private final Type PathConfigType = tf.abstractDataType(store, "PathConfig"); 
 	private final Type pathConfigConstructor = tf.constructor(store, PathConfigType, "pathConfig");
 	
-	private final List<ISourceLocation> srcs;		// List of locations to search for source files
-	private final List<ISourceLocation> libs;     // List of (library) locations to search for derived files
-	private final List<ISourceLocation> ignores; 	// List of (library) locations to ignore while compiling
-	private final List<ISourceLocation> javaCompilerPath;     // List of (library) locations to use for the compiler path of generated parsers
-	private final List<ISourceLocation> classloaders;     // List of (library) locations to use to bootstrap classloaders from
-    
-	private final ISourceLocation bin;  // Global location for derived files outside projects or libraries
+	private final List<ISourceLocation> srcs;		
+	private final List<ISourceLocation> libs;     
+    private final ISourceLocation bin;  
+	private final List<ISourceLocation> ignores; 	
+	private final ISourceLocation generatedSources;     
+	private final List<IConstructor> messages;     
+	
 
-	private static ISourceLocation defaultStd;
-	private static List<ISourceLocation> defaultIgnores;
-	private static List<ISourceLocation> defaultJavaCompilerPath;
-	private static List<ISourceLocation> defaultClassloaders;
-	private static ISourceLocation defaultBin;
+	// defaults are shared here because they occur in different use places.
+	private static final List<ISourceLocation> defaultIgnores = Collections.emptyList();
+	private static final ISourceLocation defaultGeneratedSources = URIUtil.rootLocation("unknown");
+	private static final List<IConstructor> defaultMessages = Collections.emptyList();
+	private static final ISourceLocation defaultBin = URIUtil.rootLocation("unknown");
```

Comment:
`URIUtil::invalidLocation` exists for this. (also 2 lines up)

Reply:
unknown is not invalid. I will rename the method and the field.